### PR TITLE
Add user setting to specify max number of references in panel

### PR DESCRIPTION
**Pull request by mrnugget**
This is a first step towards fixing #28480 by allowing users to overwrite how many references are shown in the panel.

Also tagging @sourcegraph/frontend-platform here because I touched some components here.

https://user-images.githubusercontent.com/1185253/149158840-76b76c3e-4af2-441c-8507-d23285e8cf54.mp4

### CODE REVIEW

File: client/web/src/repo/blob/panel/BlobPanel.tsx [hunk 1]
```diff
@@ -97,6 +102,7 @@ export function useBlobPanelViews({
                         selector: null,
                         priority,
 
+                        maxLocationResults: id === 'references' ? maxReferences : undefined,
```
Comment (mrnugget) [inline client/web/src/repo/blob/panel/BlobPanel.tsx:105]: Note that this hardcoding here mirrors what we do ~10 lines further down. So I assumed it's "okay" to add this special case here.

Comment (efritz) [reply client/web/src/repo/blob/panel/BlobPanel.tsx:105]: Do we not want to limit definitions and interfaces as well, though?

File: schema/settings.schema.json [hunk 2]
```diff
@@ -299,6 +299,10 @@
       "minimum": 0,
       "default": 0
     },
+    "codeIntelligence.maxReferences": {
```
Comment (efritz) [inline schema/settings.schema.json:302]: Let's add a changelog item noting this.

Comment (efritz) [reply schema/settings.schema.json:302]: We'll also have to make sure we can raise the limit on the extension, which hard-codes the number of results from the API: https://github.com/sourcegraph/code-intel-extensions/pull/737

**Review by efritz — APPROVED** [review]

File: client/web/src/repo/blob/panel/BlobPanel.tsx (hunk 1 above)
Comment (macraig) [reply client/web/src/repo/blob/panel/BlobPanel.tsx:105]: Yes, the original ticket was for references specifically. 

I understand that we don't limit definitions (which should be fine as you shouldn't have more than one?). For implementations, it feels like we could mimic references as it is (sort of) a list of where a function has been implemented. What do you think?

Comment (mrnugget) [reply client/web/src/repo/blob/panel/BlobPanel.tsx:105]: Fine with me! We can rename the setting to `codeIntelligence.maxPanelResults` and use it for all three, so it's consistent. Will implement.

Comment (mrnugget) [reply client/web/src/repo/blob/panel/BlobPanel.tsx]: Renamed the setting, use it for `def` and `references` now. For `implementations` we have to change the extension. Will look into that.

File: schema/settings.schema.json (hunk 2 above)
Comment (mrnugget) [reply schema/settings.schema.json]: Changelog item added. Other point discussed in Slack.

**Pull request merged**